src/window/sound_options.c

```diff
@@ -144,12 +144,16 @@ static void button_ok(int param1, int param2)
 
 static void button_cancel(int param1, int param2)
 {
+    int music_sound_is_dirty = setting_sound_is_enabled(SOUND_MUSIC) != data.original_music.enabled;
```

Comment:
This variable is only needed within the `if (data.original_music.enabled)` block. Please move it there.